pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -540,6 +541,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 		osdProps.placement.ApplyToPodSpec(&deployment.Spec.Template.Spec)
 	}
 
+	// Change TCMALLOC_MAX_TOTAL_THREAD_CACHE_BYTES if the OSD has been annotated with a value
+	osdAnnotations := cephv1.GetOSDAnnotations(c.spec.Annotations)
+	tcmallocMaxTotalThreadCacheBytes, ok := osdAnnotations[tcmallocMaxTotalThreadCacheBytesEnv]
```

Comment:
Annotations are just intended to be annotations on the pods. I don't think we should use annotations to have other side effects. If we need to expose a setting for OSDs, it has always been in the `storage` section of the cluster CR. 

This is such a specific setting, couldn't this setting be applied with the ceph.conf overrides just like any other advanced ceph setting? Why do we need to set it as an env var on the osd pod? As another alternative, couldn't they set this in the mon store?

Reply:
It's not an OSD setting. It is a tcmalloc option so it must be part of the OSD environment variables when running. So it must be part of the execution context of the OSD process. The `storage` section really seems overkill for that simple purpose. It's not something we tune every day, so IMO not worth the option. That's why I've used annotations which are much more flexible.
What kind of side effects are you thinking about?

As per above, ceph.conf is not possible nor mon store, it's not an OSD setting.

Reply:
Got it, this isn't a ceph setting in any way, the only option is an env var on the osd pod.

Reply:
If we find other settings that need to be applied similarly, we might consider doing something in the CR schema. For now, sounds good to use an annotation since it's not a setting we expect anyone to override. The default we read from the file should cover most cases.